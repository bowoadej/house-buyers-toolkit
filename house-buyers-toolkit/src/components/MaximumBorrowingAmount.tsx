import React, { useState } from 'react'
import Disclaimer from './Disclaimer';

export default function MaximumBorrowingAmount() {
    const [NumberofBuyers, setNumberofBuyers] = useState(0);
    const [showBorrowingAmount, setshowBorrowingAmount] = useState(false);
    const [isActive, setIsActive] = useState(false);


    const [calculatedAmountArray, setcalculatedAmountArray] = useState<{ firstAmount: Number; secondAmount: Number; }>({
        firstAmount: 0,
        secondAmount: 0,
    })

    function settingNumberOfBuyers(buyers: number) {
        setNumberofBuyers(buyers)
        console.log(NumberofBuyers)
    }

    function handleClick() {
        const numberOfBuyers: number = NumberofBuyers;

        if (numberOfBuyers == 1) {
            const firstPersonIncome: number = parseFloat((document.getElementById("firstPersonIncome") as HTMLInputElement).value);

            const firstCalculatedAmount = (firstPersonIncome) * 3.5;

            const secondCalculatedAmount = (firstPersonIncome) * 5
            setcalculatedAmountArray({ firstAmount: firstCalculatedAmount, secondAmount: secondCalculatedAmount });
            setshowBorrowingAmount(true);
            return calculatedAmountArray;

        } else {
            const firstPersonIncome: number = parseFloat((document.getElementById("firstPersonIncome") as HTMLInputElement).value);
            const secondPersonIncome: number = parseFloat((document.getElementById("secondPersonIncome") as HTMLInputElement).value) || 0;

            const firstCalculatedAmount = (firstPersonIncome + secondPersonIncome) * 3.5;

            const secondCalculatedAmount = (firstPersonIncome + secondPersonIncome) * 5
            setcalculatedAmountArray({ firstAmount: firstCalculatedAmount, secondAmount: secondCalculatedAmount });
            setshowBorrowingAmount(true);
            return calculatedAmountArray;

        }






    }

    function toggleActive() {
        if (isActive) {
            setIsActive(false);
        } else {
            setIsActive(true);
        }
    };

    return (
        <div className="accordion-maximum-borrowing-amount" >
            <div className="accordion-item-maximum-borrowing-amount" >
                <div data-testid="accordionTitle" className='accordion-title' onClick={() => toggleActive()}><h2>Maximum Borrowing Amount <div className='accordion-icon'>{isActive ? '-' : '+'}</div></h2></div>
                {isActive &&
                    <div className='maximum-borrowing-amount-calculator'>
                        <div className='maximum-borrowing-amount'>
                            <div id='no-of-buyers'>
                                <h3>How Many Buyers Are Involved?</h3>
                                <label>1</label>
                                <input type='radio' name="number_of_buyers" id='numberOfBuyers' value="1" onClick={() => settingNumberOfBuyers(1)} />
                                <br></br>
                                <label>2</label>
                                <input type='radio' name="number_of_buyers" id='numberOfBuyers' value="2" onClick={() => settingNumberOfBuyers(2)} />
                            </div>
                        </div>
                        <div id='person-one-income'>
                            <h3>First persons income</h3>
                            <input type='number' id="firstPersonIncome" />
                        </div>
                        {NumberofBuyers == 2 &&
                            < div id='person-two-income'>
                                <h3>Second Person's Income</h3>
                                <input type='number' id='secondPersonIncome' />
                            </div>
                        }
                        <Disclaimer />
                        <div className='submit'>
                            <button onClick={handleClick}>Submit</button>
                        </div>
                        {showBorrowingAmount &&
                            <div className='maxBorrowingAmountResult'>
                                <p>The Estimated Maximum Borrowing Amount is £{calculatedAmountArray.firstAmount.toString()} - £{calculatedAmountArray.secondAmount.toString()}</p>
                            </div>
                        }
                    </div >
                }
            </div>
        </div>
    )
}