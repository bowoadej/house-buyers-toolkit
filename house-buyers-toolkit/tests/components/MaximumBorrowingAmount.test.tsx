import renderer from 'react-test-renderer';
import { describe, expect, it, test } from '@jest/globals';
import MaximumBorrowingAmount from '../../src/components/MaximumBorrowingAmount';

/*jest.mock('../src/components/MaximumBorrowingAmount', () => {
    return { MaximumBorrowingAmount };
});*/

it('Check Maximum Borrowing Amount Calculator component enders correctly', () => {
    const tree = renderer
        .create(<MaximumBorrowingAmount />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('testing Maximum Borrowing Amount test', () => {
    test('empty string should result in zero', () => {
        expect(MaximumBorrowingAmount).toBeDefined();
    });
});

