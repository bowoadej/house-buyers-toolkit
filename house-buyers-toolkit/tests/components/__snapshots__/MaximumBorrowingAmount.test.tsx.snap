// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Maximum Borrowing Amount Calculator component enders correctly 1`] = `
<div
  className="accordion-maximum-borrowing-amount"
  data-testid="maximum-amount-container"
>
  <div
    className="accordion-item-maximum-borrowing-amount"
  >
    <div
      className="accordion-title"
      data-testid="accordionTitle"
      onClick={[Function]}
    >
      <h2>
        Maximum Borrowing Amount 
        <div
          className="accordion-icon"
        >
          +
        </div>
      </h2>
    </div>
  </div>
</div>
`;
